core/src/main/java/com/yahoo/ycsb/generator/ExponentialLongGenerator.java

```diff
@@ -0,0 +1,98 @@
+ * implied. See the License for the specific language governing                                                                                                                    
+ * permissions and limitations under the License. See accompanying                                                                                                                 
+ * LICENSE file.                                                                                                                                                                   
+ */
+
+package com.yahoo.ycsb.generator;
+
+import com.yahoo.ycsb.Utils;
+
+/**
+ * A generator of an exponential distribution. It produces a sequence
+ * of time intervals (integers) according to an exponential
+ * distribution.  Smaller intervals are more frequent than larger
+ * ones, and there is no bound on the length of an interval.  When you
+ * construct an instance of this class, you specify a parameter gamma,
+ * which corresponds to the rate at which events occur.
+ * Alternatively, 1/gamma is the average length of an interval.
+ */
+public class ExponentialLongGenerator extends LongGenerator
+{
+    // What percentage of the readings should be within the most recent exponential.frac portion of the dataset?
+    public static final String EXPONENTIAL_PERCENTILE_PROPERTY="exponential.percentile";
+    public static final String EXPONENTIAL_PERCENTILE_DEFAULT="95";
+
+    // What fraction of the dataset should be accessed exponential.percentile of the time?
+    public static final String EXPONENTIAL_FRAC_PROPERTY = "exponential.frac";
+    public static final String EXPONENTIAL_FRAC_DEFAULT  = "0.8571428571";  // 1/7
+
+	/**
+	 * The exponential constant to use.
+	 */
+	double _gamma;
+
+	/******************************* Constructors **************************************/
+
+	/**
+	 * Create an exponential generator with a mean arrival rate of
+	 * gamma.  (And half life of 1/gamma).
+	 */
+	public ExponentialLongGenerator(double mean)
+	{
+		_gamma = 1.0/mean;
+	}
+	public ExponentialLongGenerator(double percentile, double range)
+	{
+		_gamma = -Math.log(1.0-percentile/100.0) / range;  //1.0/mean;
+	}
+
+	/****************************************************************************************/
+	
+	/** 
+	 * Generate the next item. this distribution will be skewed toward lower integers; e.g. 0 will
+	 * be the most popular, 1 the next most popular, etc.
+	 * @return The next item in the sequence.
+	 */
+	public int nextInt()
+	{
+		return (int)nextLong();
+	}
+
+	/**
+	 * Generate the next item as a long.
+	 * 
+	 * @return The next item in the sequence.
+	 */
+    @Override
+	public long nextLong()
+	{
+		return (long) (-Math.log(Utils.random().nextDouble()) / _gamma);
+	}
+
+	@Override
+	public double mean() {
+		return 1.0/_gamma;
+	}
+    public static void main(String args[]) {
```

Comment:
do not add main() methods for testing, instead add as a junit test if you want to include.